obp-api/src/test/scala/code/util/JsonUtilsTest.scala

```diff
@@ -298,4 +298,86 @@ class JsonUtilsTest extends FlatSpec with Matchers {
     val str2 = json.prettyRender(expectListResult)
     str1 shouldEqual str2
   }
+
+  "$root name field" should "properly be converted" taggedAs JsonUtilsTag in {
+    val schema = json.parse(
+      """
+        |{
+        | "$root":{
+        |     "bankId":{
+        |      "value":"id"
+        |    },
+        |    "shortName":"short_name",
+        |    "fullName":"full_name"
+        | }
+        |}
+        |""".stripMargin)
+    val sourceValue = json.parse(
+      """
+        |{
+        |    "id":"dmo.01.uk.uk",
+        |    "short_name":"uk",
+        |    "full_name":"uk"
+        |}""".stripMargin)
+
+    val expectedJson =
+      """{
+        |  "bankId":{
+        |    "value":"dmo.01.uk.uk"
+        |  },
+        |  "shortName":"uk",
+        |  "fullName":"uk"
+        |}""".stripMargin
+    val resultJson = buildJson(sourceValue, schema)
+
+    val str1 = json.prettyRender(resultJson)
+
+    str1 shouldEqual expectedJson
+
+  }
+  "$root[] name field and subField[][] type field" should "properly be converted" taggedAs JsonUtilsTag in {
+    val schema = json.parse(
+      """
+        |{
+        | "$root[]":{
+        |     "bankId[]":{
+        |      "value":"id"
+        |    },
+        |    "shortName[]":"shortName",
+        |    "fullName[]":"fullName",
+        |    "nationalIdentifier[][]":"swiftBic"
```

Comment:
pay attention `nationalIdentifier[][]`, it will convert non-array to a single item array.